lib/rails/html/scrubbers.rb

```diff
@@ -150,7 +150,7 @@ def scrub_attribute(node, attr_node)
             Loofah::HTML5::Scrub.scrub_attribute_that_allows_local_ref(attr_node)
           end
 
-          if Loofah::HTML5::SafeList::SVG_ALLOW_LOCAL_HREF.include?(node.name) && attr_name == "xlink:href" && attr_node.value =~ /^\s*[^#\s].*/m
+          if Loofah::HTML5::SafeList::SVG_ALLOW_LOCAL_HREF.include?(node.name) && attr_name == "xlink:href" && attr_node.value =~ /^\s*[^#].*/m
```

Comment:
I think the meaning of the regexp is changed. It began to match to any character that starts with `/\s/`.

I'm not sure what the original regexp want to match to. maybe:
- `/^[^\S\n]*[^#\s]/`: nearly or identical to the original regexp
- `/\A\s*[^#\s]/`: matches to `"\n\n a..."` but doesn't match to `" #\n #\n a"`

or perhaps using `attr_node.value !~ allow_pattern` would be more simple?

Reply:
This regular expression originally came from the `html5lib` library:

https://github.com/html5lib/html5lib-ruby/blob/master/lib/html5/sanitizer.rb#L141-L143

The goal of this line is to accept "local links" (like `#some-id`) but not full URLs, with initial whitespace being ignored. There are test cases for this behavior in Loofah, which rails-html-sanitizer originally copied this code from.

Reply:
So @tompng is correct, this patch changes the behavior in unwanted ways. One example test case is:

```html
<animatemotion xlink:href='\n#foo'>/</animatemotion>
```

Specifically, in this case the `xlink:href` attribute should be removed but is not because:

```ruby
val = "\n#foo"
# => "\n#foo"

val =~ /^\s*[^#\s].*/
# => nil

val =~ /^\s*[^#].*/
# => 0
```